runtime/firecrackeroci/annotation.go

```diff
@@ -0,0 +1,39 @@
+// Copyright 2018-2019 Amazon.com, Inc. or its affiliates. All Rights Reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License"). You may
+// not use this file except in compliance with the License. A copy of the
+// License is located at
+//
+//	http://aws.amazon.com/apache2.0/
+//
+// or in the "license" file accompanying this file. This file is distributed
+// on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either
+// express or implied. See the License for the specific language governing
+// permissions and limitations under the License.
+
+package firecrackeroci
+
+import (
+	"context"
+
+	"github.com/containerd/containerd/containers"
+	"github.com/containerd/containerd/oci"
+)
+
+const (
+	// VMIDAnnotationKey is the key specified in an OCI-runtime config annotation section
+	// specifying the ID of the VM in which the container should be spun up.
+	VMIDAnnotationKey = "aws.firecracker.vmID"
```

Comment:
I think annotations are meant to be DNS-style, so all lowercase.  Maybe "aws.firecracker.vm-id" or "aws.firecracker.vm.id"?

Also, per https://github.com/containerd/containerd/pull/3004 there will be support for `io.containerd.runc.v2.group=<group>` and `io.kubernetes.cri.sandbox-id=<group>` to indicate grouping in containerd.  Do we want to try using those now, or start off with our own annotation and migrate to this later?

Reply:
Ah good call, will update this to `aws.firecracker.vm.id` for now.

As for whether we should re-use the keys defined in containerd's implementation instead, that's an interesting question.

On one hand, it would be nice to automatically group containers into VMs using these already defined annotations so users can switch to firecracker-containerd more seamlessly.

On the other hand, I don't know if we want to entangle ourselves and our users with keys that have meaning out of our control. I.e. what if some other random part of a user's stack starts also interpreting `io.kubernetes.cri.sandbox-id` or `io.containerd.runc.v2.group` but does something the user doesn't want? That user would probably prefer to just have a key like `aws.firecracker.vm.id` so they can specify the firecracker VM without any extra side-effects.

Maybe what'd be best is for our shim implementation to interpret each one with some priority, i.e. use `aws.firecracker.vm.id` if set, if not check `io.kubernetes.cri.sandbox-id`, if not check another, etc.

I'll make an issue for following up on this.

Reply:
https://github.com/firecracker-microvm/firecracker-containerd/issues/162